kiwipy/rmq/threadcomms.py

```diff
@@ -348,11 +348,11 @@ def no_reply(self) -> bool:
     def state(self) -> str:
         return self._task.state
 
-    def process(self) -> ThreadFuture:
-        return aiothreads.aio_future_to_thread(self._task.process())
+    async def process(self) -> ThreadFuture:
```

Comment:
I think the tricks here make the `process` and `requeue` blew the async functions. I don't have a clear view of how these two functions used in plumpy/aiida and why there can be async functions.

Reply:
The `process` change is not necessary, I will revert it. But I think the `requeue` one is necessary since it has to call `nack` which is now async, isn't it?

Reply:
Let's see if reverting `process` back to regular function still works or not. But I think you are right.

Reply:
Wait, in the latest commit, you revert the `process` of `RmqIncomingTask`, what I mentioned here is `process` and `requeue` of `RmqThreadIncomingTask`.
I think the `requeue` of `RmqThreadIncomingTask` never be called, the `del` will trigger the GC and call `requeue` from `RmqIncomingTask`.

Reply:
Ah yeah, I forgot about `RmqThreadIncomingTask.process`. Reverted that now. However, I don't understand why `RmqThreadIncomingTask.requeue` doesn't have to be async. I reverted that as well and the tests seem to run. Is it simply never called like you seem to suggest? If so, why not?

Reply:
What I understand is the `requeue` is called by `_outcome_destroyed` which is only registered by `RmqIncomingTask` but not `RmqThreadIncomingTask`. It will be called only if the reference to the `outcome` is all destroyed that GC is being triggered. 
I think after you revert them to regular functions all makes sense, and it was I tried to implement after first discussion with @muhrin. Maybe I didn't do the clear-up that fail the test or maybe `aio-pika` 8.2 fix something wrong with `8.1.8`.